src/stacked_borrows/stack.rs

```diff
@@ -48,7 +48,7 @@ pub struct Stack {
 #[cfg(feature = "stack-cache")]
 #[derive(Clone, Debug)]
 struct StackCache {
-    tags: [SbTag; CACHE_LEN], // Hot in find_granting
+    items: [Item; CACHE_LEN], // Hot in find_granting
```

Comment:
Oh, and cache lookup just masks the item to its tag... clever.

Reply:
Yes :)
The ability to do this synergizes very nicely with the change in https://github.com/rust-lang/miri/pull/2342, because now a lot of operations only ever touch the tip of the stack instead of doing near-random access into the middle of it.

Reply:
I don't quite understand the relationship to https://github.com/rust-lang/miri/pull/2342. That also only did local accesses to where the thing got inserted?

Reply:
You're right, this shouldn't make much of a difference, but I'm pretty sure it does. At a guess, the access pattern that `Vec::insert` ends up doing hides the latency of the CPU cache miss here. :shrug: Or maybe I got too excited about this cleanup and misattributed some of the speed improvement of just dropping the memory footprint to this change.